src/node.cc

```diff
@@ -1513,7 +1513,14 @@ static void ReportException(Environment* env,
         name->IsUndefined()) {
       // Not an error object. Just print as-is.
       node::Utf8Value message(env->isolate(), er);
-      PrintErrorString("%s\n", *message);
+
+      // If an error occurs while creating message, the length will be zero,
+      // but PrintErrorString() will display garbage characters. So, if the
+      // length is zero, just print a newline.
+      if (message.length() == 0)
+        PrintErrorString("\n");
+      else
+        PrintErrorString("%s\n", *message);
```

Comment:
Ah, I missed that it's using `node::Utf8Value` (which indeed never returns a nullptr because it returns a pointer to its embedded char array.)

If you use `String::Utf8Value`, you should see `*message == nullptr` after a .toString() exception.